src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -273,19 +273,44 @@ class GenerateHierarchyFunctor
                   : const_cast<Node *>(&(_leaf_nodes(i - n))));
   }
 
+  KOKKOS_FUNCTION int calculateRope(int range_right, int delta_right) const
+  {
+    int rope;
+    if (range_right != _num_internal_nodes)
+    {
+      // The way Karras indices constructed, the rope is going to be the right
+      // child of the first internal that we are in the left subtree of. The
+      // determination of whether that node is internal or leaf requires an
+      // additional delta() evaluation.
+      rope = range_right + 1;
+      if (delta_right < delta(range_right + 1))
+        rope += _num_internal_nodes;
```

Comment:
I am still struggling to wrap my head around that check.  Can you elaborate.  (I figured out the traverse.)

Reply:
If `delta(i+1) < delta(i)`, then you know that leaf node `i` is going to be a left child of some internal node. As such, you know that no rope is going to lead to it directly, as it will point to some of its predessor. On the other hand, if it is a right child of some internal node, you combine that with the fact that it's just one to the right of your current range, so it means that it's going to be the first node in the right subtree of the left subtree you are part of. 

In the figure in the description, this happens to leaf nodes 2 and 7.